lib/Runtime/Library/JSON.cpp

```diff
@@ -424,85 +427,114 @@ namespace JSON
         return StrHelper(key, value, holder);
     }
 
-    Js::Var StringifySession::Str(Js::JavascriptString* key, Js::PropertyId keyId, Js::Var holder)
+    Js::Var StringifySession::Str(Js::JavascriptString* key, Js::PropertyId keyId, Js::Var holder, Js::Var value)
     {
-        Js::Var value = nullptr;
         // We should look only into object's own properties here. When an object is serialized, only the own properties are considered,
         // the prototype chain is not considered. However, the property names can be selected via an array replacer. In this case
         // ES5 spec doesn't say the property has to own property or even to be enumerable. So, properties from the prototype, or non enum properties,
         // can end up being serialized. Well, that is the ES5 spec word.
         //if(!Js::RecyclableObject::FromVar(holder)->GetType()->GetProperty(holder, keyId, &value))
 
-        if (VirtualTableInfo<Js::PropertyString>::HasVirtualTable(key))
+        if (value == nullptr)
```

Comment:
Just to make sure I understand this change- value is not null only for the stringification of the initial object, right? That is, from the call from Line 328 (` result = stringifySession.Str(scriptContext->GetLibrary()->GetEmptyString(), propertyId, wrapper, value);`)?

Reply:
See `Str` definition on JSON.h file. `value = nullptr` default argument. Some of the consumers of `Str` knows the `value` hence no need to lookup for it once again